Date: Tue, 16 Mar 2004 22:18:28 +0100                                                                                                                               
Subject: Re: [linux-usb-devel] Usb patch : interval of iso urb doesn't get                                                                                          
        initialized.                                                                                                                                                
From: areversat <areversat@tuxfamily.org>                                                                                                                           
To: Olaf Hering <olh@suse.de>                                                                                                                                       

This patch is still required as far as i know. The eciadsl driver
doesn't work with the 2.6.4 kernel without that patch. I don't think
anybody has tested it with the 2.6.5rc1 kernel. If this is needed i can
do it next week-end (not sooner).

Antoine REVERSAT aka Crevetor


--- linux-2.6.5-rc1/drivers/usb/core/devio.c~	2003-12-05 23:42:24.000000000 +0100
+++ linux-2.6.5-rc1/drivers/usb/core/devio.c	2004-03-16 22:23:05.000000000 +0100
@@ -862,7 +862,7 @@ static int proc_submiturb(struct dev_sta
 		uurb.number_of_packets = 0;
 		if (!(ep_desc = usb_epnum_to_ep_desc(ps->dev, uurb.endpoint)))
 			return -ENOENT;
-		interval = ep_desc->bInterval;
+		interval = (1 << (ep_desc->bInterval - 1));
 		if (uurb.buffer_length > 16384)
 			return -EINVAL;
 		if (!access_ok((uurb.endpoint & USB_DIR_IN) ? VERIFY_WRITE : VERIFY_READ, uurb.buffer, uurb.buffer_length))
